hs_core/views/__init__.py

```diff
@@ -697,3 +697,18 @@ def _set_resource_sharing_status(request, user, resource, is_public, is_discover
         # set isPublic metadata AVU accordingly
         istorage = IrodsStorage()
         istorage.setAVU(resource.short_id, "isPublic", str(resource.raccess.public))
+
+
+def _get_message_for_setting_resource_flag(resource, resource_flag):
```

Comment:
What I am suggesting here is perhaps there is a better way to do the two checks one time and know whether both checks have passed and which check fails otherwise. Since can_be_public() already do both checks but only returns true or false, so you don't know whether content files are missing or required metadata are missing if it returns false, then you call the two checks again to determine this subsequently to show informative messages. Currently these two checks are simple and not expensive at all, but there might be a possibility it gets more complicated down the road when we have more required metadata elements for example, so it'd be good if we can remove duplicate checks, which is the main reason I made all these related suggestions for your reference. It is not a roadblock at all, so If you think it is not an issue at this point, I am totally fine with it.